src/storage/secondary/block/block_index_builder.rs

```diff
@@ -83,4 +87,8 @@ impl BlockIndexBuilder {
     pub fn into_index(self) -> Vec<BlockIndex> {
         self.indexes
     }
+
+    pub fn set_first_key(&mut self, first_key: Vec<u8>) {
```

Comment:
I'd prefer having the signature as

```
pub fn set_first_key(&mut self, first_key: &[u8]) {
  self.first_key.clear();
  self.first_key.extend(first_key);
}
```

So as to reduce allocation as much as possible.

By the way, it seems that `first_key` will only be used on `finish_block`. Is it possible to make it a parameter of `finish_block` instead of having a separate function?

Reply:
> By the way, first_key is only useful when the column is fully sorted. So we can have a parameter on column builder whether to store the first_key or not.

Agree. But I am curious about where can we know that column need sort or not? Maybe pass from `ColumnBuilderOptions` later?

Reply:
> I'd prefer having the signature as
> 
> ```
> pub fn set_first_key(&mut self, first_key: &[u8]) {
>   self.first_key.clear();
>   self.first_key.extend(first_key);
> }
> ```
> 
> So as to reduce allocation as much as possible.
> 
> By the way, it seems that `first_key` will only be used on `finish_block`. Is it possible to make it a parameter of `finish_block` instead of having a separate function?

This means that we need to modify `PrimitiveColumnBuilder<T>::finish_builder()` interface, that's ok...